src/debug.c

```diff
@@ -1023,6 +1023,7 @@ void debugCommand(client *c) {
 
 /* =========================== Crash handling  ============================== */
 
+#ifndef VALKEY_USE_TEST_SERVER_ASSERT
```

Comment:
Didn't see another discussion, but why is this here?

Reply:
on macOS / ARM, the linker can't handle the duplicate symbol of `serverAssert` + `main` (both redefined in `test_main.c`). CMake now exposes these macros to skip these functions when building the unit tests as these functions can be resolved from `test_main.c`